frontend/lib/resolution/default-functions.cpp

```diff
@@ -407,6 +506,10 @@ getCompilerGeneratedMethodQuery(Context* context, const Type* type,
       result = generateInitCopySignature(context, compType);
     } else if (name == USTR("deinit")) {
       result = generateDeinitSignature(context, compType);
+    } else if (type->isDomainType()) {
+      result = generateDomainMethod(context, type->toDomainType(), name);
+    } else if (type->isArrayType()) {
+      result = generateArrayMethod(context, type->toArrayType(), name);
```

Comment:
Can we use `if (auto domainType = ...)` here to avoid a second `toDomainType` cast?